cmake/modules/AddPureSwift.cmake

```diff
@@ -82,6 +82,39 @@ function(_add_host_swift_compile_options name)
   endif()
 endfunction()
 
+# Set compile options for C/C++ interop
+function(_set_swift_cxx_interop_options name)
+  target_compile_options(${name} PRIVATE
+    "SHELL: -Xcc -std=c++17 -Xcc -DCOMPILED_WITH_SWIFT"
```

Comment:
Why is `__swift__` not sufficient? I'm just wondering if we can remove the `-DCOMPILED_WITH_SWIFT`.

Reply:
I guess that's enough. Let me check it in follow-ups